### ISSUE TITLE
Multiple instances of module show same thing

### ISSUE THREAD

**Issue by PreschoolGirl** [NateDee/MMM-CTA#1]
Thanks for making this! Trying to use two instances of the module, one for my el stop and one for my bus stop. But both just show the el information.

**Comment by NateDee** [comment]
Hi!  Thanks for the interest, gives me the motivation to polish the module.  So you have two instances of the module set in your config.js file of the mirror is that correct?  Could you show me an example of your config.js?  Please remove API keys so that you don't make them public.  

This situation should be addressed by changes I've been implementing, I will push them this weekend and let you know when I do.

If you have one instance of the module with your bus and train information in, the module should provide a stacked table (I need to update example pictures).  Does that work properly?

**Comment by PreschoolGirl** [comment]
You're the only one I could find making a module for Chicago busses: thanks for being the hero we need. Below is my config.js for this module. 

As you can see, I've got one instance with just a bus stop, and another instance with a bus and train stacked. Both just show the train. I'll try to get a screen shot in here. 

```javascript
                {
                        module: 'MMM-CTA',
                        position: 'bottom_right',
                        config: {
                                updateTime: 60000, 
                                ctaApiKey: '<my bus api key>',
                                busStopName: 'Lawrence Westbound',  
                                stopId: 3791, 
                                maxResult: 4,   
                                }
                        },
                        {
                        module: 'MMM-CTA',
                        position: 'bottom_left',
                        config: {
                                updateTime: 60000, 
                                ctaApiKey: '<my bus api key>',
                                busStopName: 'Lawrence Eastbound',  
                                stopId: 3759, 
                                maxResult: 4,
                                ctaApiKeyTrain: '<my train api key>',
                                trainStopName: 'Brown Line',  
                                trainStationID: 40090, 
                                }
                        },
```

**Comment by NateDee** [comment]
I was very surprised there weren't any already!  Now that I know someone besides me is interested I'll definitely make this baby shine!

So the issue is, since it's the same module, the instances will both send socket notifications with the same name "MMM-CTA-DATA".  So they will both receive the same message (there is also a bug for when you only set busses that I've fixed but haven't pushed the updates).  Good news though, we can make multi-instance work and I think it's a great feature idea to add so thank you!  I'll get to it this evening and push the necessary updates.  I will update you when I do.

**Comment by PreschoolGirl** [comment]
That's awesome! Nerds of Chicago thank you.

Not a programmer: would it make sense to have the stops be an array, so you can add as many as you want in an single instance?

**Comment by NateDee** [comment]
Hi, I've made the changes to make multiple instances possible as well as cleaned up the code a bit.  Please see the changes to the config options in the readme! I will definitely add an option to stack multiple bus stops in a single table in the near future.  Update the module and let me know if you can get it working!

**Comment by PreschoolGirl** [comment]
Exciting news! It's broken in a new way. For "instance 1" my intended behavior is to just have a bus stop. Now that's just a blank corner (where before it was a copy of the train from the other instance). For "instance 2" my intended behavior is to have both a bus and a train. Now it's just the train. 

I found this in my logs: 
```
1|mm       | Launching application.
1|mm       | MMM-CTA: Train config incomplete!!!
1|mm       | Initialize button monitor_control on PIN 25
1|mm       | MMM-CTA-DATA2
1|mm       | CTA request fired train.
1|mm       | { ctatt: 
1|mm       |    { tmst: '2018-03-23T21:38:44',
1|mm       |      errCd: '0',
1|mm       |      errNm: null,
1|mm       |      eta: [ [Object], [Object], [Object], [Object] ] } }
1|mm       | Shutting down server...
1|mm       | Stopping module helper: updatenotification
1|mm       | Stopping module helper: MMM-CTA
```

And here's my updated config.js (where I probably am doing something wrong, since, again, I have no idea what I'm doing):

```javascript
                        {
                        module: 'MMM-CTA',
                        position: 'bottom_right',
                        config: {
                                updateTime: 60000, 
                                ctaApiKey: 'GJQ4aMmb2dXkd9RbCNUTTpkC5',
                                busStopName: 'Lawrence Westbound',  
                                stopId: 3791, 
                                maxResult: 4,
                                moduleInstance: 1,   
                                }
                        },
                        {
                        module: 'MMM-CTA',
                        position: 'bottom_left',
                        config: {
                                updateTime: 60000, 
                                ctaApiKey: 'GJQ4aMmb2dXkd9RbCNUTTpkC5',
                                busStopName: 'Lawrence Eastbound',  
                                stopId: 3759, 
                                maxResult: 4,
                                ctaApiKeyTrain: 'e4457d3281cc477488fa0fcd66a76$
                                trainStopName: 'Brown Line',  
                                trainStationID: 40090,
                                maxResultTrain: 4,
                                moduleInstance: 2, 
                                }
                        },
```

**Comment by PreschoolGirl** [comment]
Crap, just realized I commented without deleting my API keys and I can't edit. Let's delete and try again. 

So now it's broken in a different way. I'm trying to get "instance 1" to be just the bus, and "instance 2" to be a bus and train. Before, they were both just the train from "instance 2." Now,  "instance 1" is missing entirely, and "instance 2" is just the train (no bus).

Here's something I found from my logs:
```1|mm       | Launching application.
1|mm       | MMM-CTA: Train config incomplete!!!
1|mm       | Initialize button monitor_control on PIN 25
1|mm       | MMM-CTA-DATA2
1|mm       | CTA request fired train.
1|mm       | { ctatt: 
1|mm       |    { tmst: '2018-03-23T21:38:44',
1|mm       |      errCd: '0',
1|mm       |      errNm: null,
1|mm       |      eta: [ [Object], [Object], [Object], [Object] ] } }
1|mm       | Shutting down server...
1|mm       | Stopping module helper: updatenotification
1|mm       | Stopping module helper: MMM-CTA
1|mm       | Stopping module helper: MMM-Buttons
```

And here's the relevant portion of my config.js file (where, again I've probably screwed something up because I have no idea what I'm doing):
```javascript
                       {
                        module: 'MMM-CTA',
                        position: 'bottom_right',
                        config: {
                                updateTime: 60000, 
                                ctaApiKey: '<my bus API key>',
                                busStopName: 'Lawrence Westbound',  
                                stopId: 3791, 
                                maxResult: 4,
                                moduleInstance: 1,   
                                }
                        },
                        {
                        module: 'MMM-CTA',
                        position: 'bottom_left',
                        config: {
                                updateTime: 60000, 
                                ctaApiKey: '<my bus API key>',
                                busStopName: 'Lawrence Eastbound',  
                                stopId: 3759, 
                                maxResult: 4,
                                ctaApiKeyTrain: '<my train api key>',
                                trainStopName: 'Brown Line',  
                                trainStationID: 40090,
                                maxResultTrain: 4,
                                moduleInstance: 2, 
                                }
                        },
```

**Comment by NateDee** [comment]
I see, so it looks like the bus request is not firing.  I may have had an incorrect busUrl in the script.  Update one more time, keep your config the same. Let me know if that solves it.

To explain those logs:  The "Train config incomplete" is okay, since in one instance you do not have the train config set.  But the DATA2 instance is only receiving train data, which tells me something is wrong with the bus request.  Looking at the busUrl, it's different then what I have set in my config (which works).  So it should be the url, which I just updated, and I suspect should fix the issues.
